src/bsdutils/meson.build

```diff
@@ -1,4 +1,27 @@
+code = '''#include <stdio.h>
+#include <fts.h>
+
+int main(void)
+{
+  char *const fts_args[] = { ".", NULL };
+  FTS *tree = fts_open(fts_args, FTS_SEEDOT, 0);
+  if (tree)
+    return fts_close(tree);
+  return -1;
+}'''
+
+dep_musl_fts = []
+
+if cc.has_header_symbol('fts.h', 'fts_open', required: false)
+  if not cc.links(code, name : 'fts link check')
+    dep_musl_fts = dependency('musl-fts', required: true)
```

Comment:
@eli-schwartz Now the only potential problem I see here is that I'm not doing another links() check. But I'm not sure I want to worry about other, unknown future cases yet. AFAIK, I only have to worry about **musl-fts** failing cc.link() after it passes cc.has_header_symbol(). That sound about right to you?

Reply:
The issue is that there are three cases:
1. fts doesn't exist all, period. No header, no symbol
2. fts.h exists, and is usable without adding any libraries
3. fts.h and an external library exist, and you need to use it, and dependency() is the best way to do that

The only way to detect case 2 is to use cc.links(). You don't actually have to use cc.has_header() since the links test with suitable test code already tries to `#include <fts.h>`.

Reply:
I see. I'll do some re-writing again later.